File: 4-clustered-eventbus/src/main/java/Producer.java
import io.vertx.core.AbstractVerticle;
import io.vertx.core.Vertx;
import io.vertx.core.VertxOptions;
import io.vertx.core.json.JsonObject;

/**
 * @author <a href="http://escoffier.me">Clement Escoffier</a>
 */
public class Producer {

  public static void main(String[] args) {
    Vertx.clusteredVertx(new VertxOptions(),
        ar -> {
          Vertx vertx = ar.result();
          vertx.setPeriodic(1000,
              l -> {
                vertx.eventBus().publish("events",
                        new JsonObject()
                            .put("message", "hello")
                            .put("from", "java producer"));
      });
    });
  }
}

File: 7-polyglot/src/main/js/producer.js
var Random = java.util.Random;
var random = new Random();

var eb = vertx.eventBus();
vertx.setPeriodic(1000,
    function (v) {
        var val = random.nextInt(60);
        console.log("Javascript -> Publishing " + val);
        eb.publish("data", {
            "value": val,
            "id": "javascript"
        });
    }
);